shell/client/grainview.js

```diff
@@ -401,9 +410,9 @@ GrainView.prototype.sessionStatus = function () {
 GrainView.prototype.route = function () {
   this._dep.depend();
   if (this._token) {
-    return "/shared/" + this._token;
+    return "/shared/" + this._token + this._path;
```

Comment:
We should probably also include the query string and fragment in both of these too?